src/com/spikes2212/genericsubsystems/BasicSubsystem.java

```diff
@@ -0,0 +1,79 @@
+package com.spikes2212.genericsubsystems;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.Subsystem;
+
+/**
+ * This class represents a subsystem that moves with a limitation which depends
+ * on the speed.
+ *
+ * @author Omri "Riki" Cohen
+ */
+public class BasicSubsystem extends Subsystem {
+
+	/**
+	 * This function should check if the subsystem can move given the speed
+	 */
+	public final Function<Double, Boolean> canMove;
+	private Consumer<Double> speedConsumer;
+
+	/**
+	 * this constructs a new {@link BasicSubsystem} subsystem.
+	 * 
+	 * @param speedConsumer
+	 *            the thing using the speed (usually a motor/ motors).
+	 * @param canMove
+	 *            the limitation on the movement, which depends on the speed.
+	 */
+	public BasicSubsystem(Consumer<Double> speedConsumer, Function<Double, Boolean> canMove) {
+		this.canMove = canMove;
+		this.speedConsumer = speedConsumer;
+	}
+
+	/**
+	 * this constructs a new {@link BasicSubsystem} subsystem without any
+	 * limitations.
+	 * 
+	 * @param speedConsumer
+	 *            the thing using the speed (usually a motor/ motors).
+	 */
+	public BasicSubsystem(Consumer<Double> speedConsumer) {
+		this(speedConsumer, (s) -> true);
+	}
+
+	/**
+	 * Moves this subsystem with the given speed.
+	 *
+	 * @param speed
+	 *            the speed to move with.
+	 */
+	public void move(double speed) {
+		if (canMove.apply(speed))
+			speedConsumer.accept(speed);
+	}
+
+	/**
+	 * Stops this subsystem's movement.
+	 */
+	public void stop() {
+		move(0);
+	}
+
+	/**
+	 * Sets the default command. If this is not called or is called with null,
+	 * then there will be no default command for the subsystem.
+	 * 
+	 * @see edu.wpi.first.wpilibj.command.Subsystem#setDefaultCommand(edu.wpi.first.wpilibj.command.Command)
+	 */
+	public void setDefaultCommand(Command defaultCommand) {

```

Comment:
This is redundant because you're just redirecting the call to the method in `super`, which is exactly what would have happened if you just omitted this function from this class.